back-end/yelpCrawler.py

```diff
@@ -0,0 +1,203 @@
+load_dotenv(dotenv_path, override=True)
+
+API_KEY = os.environ.get("API_KEY")
+
+# API constants, you shouldn't have to change these.
+API_HOST = 'https://api.yelp.com'
+SEARCH_PATH = '/v3/businesses/search'
+BUSINESS_PATH = '/v3/businesses/'  # Business ID will come after slash.
+
+# Defaults for our simple example.
+DEFAULT_TERM = 'cafes'
+LATITUDE = 37.778281
+LONGITUDE = -122.411865
+OPEN_NOW = 'false'
+SEARCH_LIMIT = 50
+DEFAULT_LOCATION = 'San Francisco, CA'
+
+def request(host, path, api_key, url_params=None):
+    """Given your API_KEY, send a GET request to the API.
+    Args:
+        host (str): The domain host of the API.
+        path (str): The path of the API after the domain.
+        API_KEY (str): Your API Key.
+        url_params (dict): An optional set of query parameters in the request.
+    Returns:
+        dict: The JSON response from the request.
+    Raises:
+        HTTPError: An error occurs from the HTTP request.
+    """
+    url_params = url_params or {}
+    url = '{0}{1}'.format(host, quote(path.encode('utf8')))
+    headers = {
+        'Authorization': 'Bearer %s' % api_key,
+    }
+    response = requests.request('GET', url, headers=headers, params=url_params)
+
+    return response.json()
+
+#def search(api_key, term, latitude, longitude, offset):
+def search(api_key, term, location, offset):
+    """Query the Search API by a search term and location.
+    Args:
+        term (str): The search term passed to the API.
+        location (str): The search location passed to the API.
+    Returns:
+        dict: The JSON response from the request.
+    """
+    url_params = {
+        'term': term.replace(' ', '+'),
+        'location': location.replace(' ', '+'),
+        #'latitude': latitude,
+        #'longitude': longitude,
+        'open_now': OPEN_NOW.replace(' ', '+'),
+        'limit': SEARCH_LIMIT,
+        'offset': offset
+    }
+    return request(API_HOST, SEARCH_PATH, api_key, url_params=url_params)
+
+
+def get_business(api_key, business_id):
+    """Query the Business API by a business ID.
+    Args:
+        business_id (str): The ID of the business to query.
+    Returns:
+        dict: The JSON response from the request.
+    """
+    business_path = BUSINESS_PATH + business_id
+
+    return request(API_HOST, business_path, api_key)
+
+# Define default place : wework civic, default wifi='free', default parking='street'
+# def getYelpData(lat=37.778264, longi=-122.411843, wifi="Free", parking="Street"):
+def getYelpData(wifi="Free", parking="Street"):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('-s', '--term', dest='term', default=DEFAULT_TERM,
+                        type=str, help='Search term (default: %(default)s)')
+    parser.add_argument('-l', '--location', dest='location', default=DEFAULT_LOCATION,
+                        type=str, help='Search term (default: %(default)s)')
+    #parser.add_argument('-la', '--longitude', dest='longitude', default=LONGITUDE,
```

Comment:
Please remove all commented lines. If some of these are necessary, please add a comment and justify your reason.